ivy/backend_handler.py

```diff
@@ -241,6 +252,58 @@ def set_backend(backend: str):
         isinstance(backend, str) and backend not in _backend_dict,
         "backend must be one from {}".format(list(_backend_dict.keys())),
     )
+
+    if len(backend_stack) == 0:
+        _load_static_backends()
+
+    # Dynamic Backend
+    from ivy.functional.ivy.gradients import _variable, _is_variable, _variable_data
+
+    variable_ids = set()  # create an empty set to store variable object ids
+    intermediate_objs = [] # create an empty list to store intermediate objects created during conversion
+    def _is_var(obj):
+
+        if isinstance(obj, ivy.Container):
+            def _map_fn(x):
+
+                x = x.data if isinstance(x, ivy.Array) else x
+                if x.__class__.__module__ in ("numpy", "jax.interpreters.xla","jaxlib.xla_extension"):
```

Comment:
Adding this check here because `is_variable()` returns `True` for numpy and JAX arrays. Basically, these 2 frameworks don't have the concept of `stateful` tensors and so they should technically not be considered as *variables*

This does mean that if you are converting variables from `TF->JAX/Numpy->Torch` for example, the variables would not get preserved as the middle backend would basically convert them to normal native arrays.

Is this the correct behavior?